### PR TITLE
State: Update preferences to use wpcom.req directly

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

This PR updates the preferences actions to use `wpcom.req` directly rather than having to use the separate wpcom-undocumented methods. It also removes those unused undocumented methods.

Part of the effort to remove `wpcom-undocumented` and make the state more modular and packageable.

#### Testing instructions

* Open any page in Calypso.
* Verify preferences are still loaded correctly by rolling over the environment badge on the bottom right.
* Change a preference, verify it's still saved correctly.

### CODE REVIEW

File: client/state/preferences/actions.js [hunk 1]
```diff
@@ -93,11 +90,8 @@ export const savePreference = ( key, value ) => ( dispatch ) => {
 		},
 	};
 
-	return wpcom
-		.undocumented()
-		.me()
-		.preferences()
-		.update( payload )
+	return wpcom.req
+		.put( '/me/preferences', payload )
```
Comment (flootr) [inline client/state/preferences/actions.js:94]: Nit: I think we could use `.post()` directly in this case, looking at the `.put()` method and the browser devtools Network tab it looks like a regular `POST` request. Not sure what's the intention here so leaving it as a nit.

**Review by flootr — APPROVED** [review]
seems to be working as expected 👍🏻

File: client/state/preferences/actions.js (hunk 1 above)
Comment (jsnajdr) [reply client/state/preferences/actions.js:94]: The server endpoints use `POST` as a method. I'm not sure if `PUT` would even work.

**Review by jsnajdr — APPROVED** [review]
:+1:

File: client/state/preferences/actions.js (hunk 1 above)
Comment (tyxla) [reply client/state/preferences/actions.js:94]: Yep. I'm just refactoring it to what we were using under the hood: https://github.com/Automattic/wp-calypso/pull/53697/files#diff-fe438575a1dba9db3a80ea27ae6e7267ae6e4345e14d35ed1fe072bd08fdb8b6L24

One step at a time.

In a subsequent PR we might want to completely get rid of `.put()` and replace it with `.post()` which essentially it already is.

**Pull request merged**